#! /bin/sh -e
## 63_update_default_server_language.dpatch, by
## László 'GCS' Böszörményi <gcs@lsc.hu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Get the default language from the config file, and do not use the
## DP: hardcoded English.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
--- mailman-2.1.5/bin/update.orig	2004-10-14 21:51:51.000000000 +0000
+++ mailman-2.1.5/bin/update	2004-10-14 22:19:25.000000000 +0000
@@ -121,7 +121,7 @@
     #
     # First, get rid of any lists/<list> template or lists/<list>/en template
     # that is identical to the global templates/* default.
-    for gtemplate in os.listdir(os.path.join(mm_cfg.TEMPLATE_DIR, 'en')):
+    for gtemplate in os.listdir(os.path.join(mm_cfg.TEMPLATE_DIR, mm_cfg.DEFAULT_SERVER_LANGUAGE)):
         # BAW: get rid of old templates, e.g. admlogin.txt and
         # handle_opts.html
         try:
@@ -155,14 +155,14 @@
                 os.unlink(os.path.join(mlist.fullpath(), gtemplate + '.prev'))
         # Match against the lists/<list>/en/* templates
         try:
-            fp = open(os.path.join(mlist.fullpath(), 'en', gtemplate))
+            fp = open(os.path.join(mlist.fullpath(), mm_cfg.DEFAULT_SERVER_LANGUAGE, gtemplate))
         except IOError, e:
             if e.errno <> errno.ENOENT: raise
         else:
             tcksum = md5.new(fp.read()).digest()
             fp.close()
             if gcksum == tcksum:
-                os.unlink(os.path.join(mlist.fullpath(), 'en', gtemplate))
+                os.unlink(os.path.join(mlist.fullpath(), mm_cfg.DEFAULT_SERVER_LANGUAGE, gtemplate))
         # Match against the templates/* template
         try:
             fp = open(os.path.join(mm_cfg.TEMPLATE_DIR, gtemplate))
